multicluster/controllers/multicluster/clusterset_controller.go

```diff
@@ -47,9 +67,50 @@ type ClusterSetReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/reconcile
 func (r *ClusterSetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	_ = r.Log.WithValues("ClusterSet", req.Namespace)
+
+	clusterSet := &multiclusterv1alpha1.ClusterSet{}
+	err := r.Get(ctx, req.NamespacedName, clusterSet)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return ctrl.Result{}, err
+		}
+		// if errors.IsNotFound(err)
+		r.Log.Info("Received ClusterSet delete", "config", clusterSet, "leader", r.IsLeader)
+		if !r.IsLeader {
+			if err := r.remoteClusterMgr.Stop(); err != nil {
+				return ctrl.Result{}, err
+			}
+		} else {
+			if err := r.inboundLeaderMgr.Stop(); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+		return ctrl.Result{}, nil
+	}
+
+	r.Log.Info("Received ClusterSet add/update", "config", clusterSet, "leader", r.IsLeader)
+
+	// Handle create or update
+	// If create
+	//   if leader, make sure the local cluster claim is part of the leader config
+	//   if not leader, make sure the local cluster claim is part of the member config
+	if err = r.validateLocalClusterClaim(clusterSet); err != nil {
+		return ctrl.Result{}, err
+	}
 
-	// your logic here
+	// it is okay to be both a leader and member
+	// if update,
+	//    if leader - nothing to do in inbound mode
+	//    if member - if leaders have changed handle accordingly, else nothing to do.
+	if !r.IsLeader {
+		err = r.updateMultiClusterSetOnMemberCluster(clusterSet)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+	} else {
+		r.createOrUpdateMultiClusterSetOnLeaderCluster(clusterSet)
```

Comment:
looks like you only create `InboundLeaderMgr` in leader cluster, then no manager can represent a local cluster, I suppose you need a new field eg:`localClusterMgr` in `ClusterSetReconciler` to represent a local member cluster itself, so other controllers can reuse it.

Reply:
I dint understand, can you please elaborate this for me?

Reply:
I was working on integrate serviceexport controller with common area, but I found there is no way to get local member cluster ID via common area, I suppose it can be initialized by ClusterSet controller after ClusterClaim is added in local cluster?

Reply:
Got it, instead of adding it in the common area which is required to access every leader, I added it in the RemoteClusterManager
	// GetLocalClusterID returns local cluster ID
	GetLocalClusterID() common.ClusterID

Can you use that one?